### DISCUSSION TITLE
Novice looking to Stream I2S to Bluetooth Headset

### DISCUSSION THREAD

**Discussion by Fishbone69** [pschatzmann/arduino-audio-tools#1343] [Q&A] [answered]
Hi.  I'm very new to this, but am messing around with different FM Receivers.  The Si4735 receives an FM signal and can send it to digital out (I2S).  Would these audio tools (or A2DP library) allow me to stream the I2S output to a set of bluetooth headphones?  If so, which stream examples should I look at?  Oh, I forgot to mention...I am using an esp32-S3 which I understand does not support Bluetooth classic and only uses Blutooth LE

Thanks!

**Comment by pschatzmann** [comment, marked as answer]
You would need to use a regular ESP32, since this is not possible with a S3 or any other subvariant.

You can look at [this example](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-basic-api/base-i2s-a2dp/base-i2s-a2dp.ino) to get you started. However most likely you will not need to do any format conversion, but you can exchange this with a VolumeStream, so that you can control the volume. Further info can be found in the Wiki!

There are other exmaples: just look for I2S-A2DP in it's name...

**Reply by Fishbone69** [reply]
Thank you so much for all of the work and knowledge you have put into these tool.  It is sincerely appreciated.

**Reply by Fishbone69** [reply]
I have invested quite a bit of time on the esp32-S3 for my project.  I will have to put this on the back burner until someone somes up with audio streaming library for BLE.  I am a hobbyest and not genius enough to do what you guys do.  But I still am thankful for all that you do!!!!!!

Fish

**Reply by malaki86** [reply]
> I have invested quite a bit of time on the esp32-S3 for my project. I will have to put this on the back burner until someone somes up with audio streaming library for BLE. I am a hobbyest and not genius enough to do what you guys do. But I still am thankful for all that you do!!!!!!
> 
> Fish

It's not a library that requires the support - it's the chip. Bluetooth Low-Energy Audio is not only a fairly new standard, it's not in many devices yet. The ESP32-H4 (released in December 2024) is supposed to support LEA.
